### PR TITLE
[AArch64] Implements paired loadstores

### PR DESCRIPTION
**Pull request by Sonicadvance1**
This doesn't implement the paired loadstores with index because it's harder to find a game that uses those

### CODE REVIEW

**Comment by phire** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/PowerPC/JitArm64/JitAsm.cpp [hunk 1]
```diff
@@ -7,10 +7,27 @@
...
+}
+static void WriteDual16(u32 value, u32 address)
+{
+	Memory::Write_U16(value, address);
+	Memory::Write_U16(value >> 16, address + 2);
+}
+static void WriteDual8(u16 value, u32 address)
+{
+	Memory::Write_U8(value, address);
+	Memory::Write_U8(value >> 8, address + 1);
```
Comment (magumagu) [inline Source/Core/Core/PowerPC/JitArm64/JitAsm.cpp]: It would be more conventional to write this as `Memory::Write_U16(Common::swap16(value), address);`

**Comment by Mullin** [comment]
@@ -33,7 +33,7 @@ void JitArm::psq_l(UGeckoInstruction inst)
 	UBFX(R12, R11, 16, 3); // Type
 	LSL(R12, R12, 2);
 	UBFX(R11, R11, 24, 6); // Scale
-	LSL(R11, R11, 2);
+	LSL(R11, R11, 3);
 
 	Operand2 off;
 	if (TryMakeOperand2(offset, off))
 @@ -84,7 +84,7 @@ void JitArm::psq_lx(UGeckoInstruction inst)
 	UBFX(R12, R11, 16, 3); // Type
 	LSL(R12, R12, 2);
 	UBFX(R11, R11, 24, 6); // Scale
-	LSL(R11, R11, 2);
+	LSL(R11, R11, 3);
 
 	if (inst.RA || update) // Always uses the register on update
 	{
 @@ -136,7 +136,7 @@ void JitArm::psq_st(UGeckoInstruction inst)
 	UBFX(R12, R11, 0, 3); // Type
 	LSL(R12, R12, 2);
 	UBFX(R11, R11, 8, 6); // Scale
-	LSL(R11, R11, 2);
+	LSL(R11, R11, 3);
 
 	Operand2 off;
 	if (TryMakeOperand2(offset, off))
 @@ -187,7 +187,7 @@ void JitArm::psq_stx(UGeckoInstruction inst)
 	UBFX(R12, R11, 0, 3); // Type
 	LSL(R12, R12, 2);
 	UBFX(R11, R11, 8, 6); // Scale
-	LSL(R11, R11, 2);
+	LSL(R11, R11, 3);
 

Cause weird screen and weird texture on somes games

**Comment by Sonicadvance1** [comment]
Did you test this entire change or just those particular lines?
I changed the table that it was pulling from in a different location.

**Comment by Mullin** [comment]
Entire change, why ?

**Comment by Sonicadvance1** [comment]
Fun thing that due to memory loadstores being a mess on ARMv7, the change in behaviour is actually more correct. Just that due to being more correct, it's showing more breakage.
So really this shouldn't be held back because of that, I just need to fix ARMv7's loadstores in a different PR.

**Comment by Mullin** [comment]
I had a doubt I recompiled everything, it's "ok" now. The bizarre textures are still there but was present before the fix. Do you know how to fix it ?

**Pull request merged**